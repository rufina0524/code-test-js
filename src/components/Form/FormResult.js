import React from 'react';
import PropTypes from 'prop-types';

const style = {
  position: 'absolute',
  left: 0,
  margin: 20,
  height: 300,
  overflowY: 'scroll'
}
const FormResult = (props) => {
  const { data, onClick, style: itemStyle } = props;

  if (data && data.length) {
    return (
      <div style={style}>
        {
          data.map(item => 
            <button
              style={itemStyle}
              key={item.id}
              id={item.id}
              onClick={onClick}
            >
              { item.name }
            </button>
          )
        }
      </div>
    );  
  }
  return null;
};

FormResult.propTypes = {
  data: PropTypes.array.isRequired,
  onClick: PropTypes.func,
  style: PropTypes.object
};

export default FormResult;