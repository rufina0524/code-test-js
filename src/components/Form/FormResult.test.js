import React from 'react';
import FormResult from './FormResult';
import renderer from 'react-test-renderer';

const mockData = [
    { id: 'mock-1' },
    { id: 'mock-2' }
];

describe('FormResult', () => {
    it('should match snapshot', () => {
    const comp = renderer
        .create(
          <FormResult
            data={mockData}
            style={{}}
            onClick={jest.fn()}
          />
        )
        .toJSON();
      expect(comp).toMatchSnapshot();
    });
});