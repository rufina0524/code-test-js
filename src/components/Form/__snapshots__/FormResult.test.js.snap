// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormResult should match snapshot 1`] = `
<div
  style={
    Object {
      "height": 300,
      "left": 0,
      "margin": 20,
      "overflowY": "scroll",
      "position": "absolute",
    }
  }
>
  <button
    id="mock-1"
    onClick={[MockFunction]}
    style={Object {}}
  />
  <button
    id="mock-2"
    onClick={[MockFunction]}
    style={Object {}}
  />
</div>
`;
